polkadot/cli/src/lib.rs

```diff
@@ -57,10 +61,41 @@ mod informant;
 use std::io;
 use std::net::SocketAddr;
 use std::path::{Path, PathBuf};
+use std::sync::Arc;
+
 use futures::sync::mpsc;
 use futures::{Sink, Future, Stream};
 use tokio_core::reactor;
+use parking_lot::Mutex;
 use service::ChainSpec;
+use primitives::block::Extrinsic;
+
+struct RpcTransactionPool {
+	inner: Arc<Mutex<txpool::TransactionPool>>,
+	network: Arc<network::Service>,
+}
+
+impl substrate_rpc::author::AuthorApi for RpcTransactionPool {
+	fn submit_extrinsic(&self, xt: Extrinsic) -> substrate_rpc::author::error::Result<()> {
+		use primitives::hexdisplay::HexDisplay;
+		use polkadot_runtime::UncheckedExtrinsic;
+		use codec::Slicable;
+
+		info!("Extrinsic submitted: {}", HexDisplay::from(&xt.0));
+		let (decoded, encoded_bytes) = xt.using_encoded(|s| {
+				UncheckedExtrinsic::decode(&mut &s[..]).map(|d| (d, s.to_vec()))
+			})
+			.ok_or(substrate_rpc::author::error::ErrorKind::InvalidFormat)?;
+		info!("Correctly formatted: {:?}", decoded);
+
+		let hash = self.inner.lock().import(decoded)
+			.map(|v| v.hash().0.into())
+			.map_err(|_| substrate_rpc::author::error::ErrorKind::PoolError)?;
+
+		self.network.on_new_transactions(&[(hash, encoded_bytes)]);
```

Comment:
this would end up propagating stale transactions; we might want to ensure `cull` is called first and propagate only after it is clear that the tx wasn't discarded.